src/standard_library/array_methods.md

```diff
@@ -0,0 +1,139 @@
+
+# Array
+
+For convenience, the STD provides some ready-to-use, common methods for arrays[^migrationNote]:
+
+## len
+
+Returns the length of an array
+
+```rust
+fn len(_array: Self) -> comptime Field
+```
+
+example
+
+```rust
+// arr = [42, 42]
+fn main(arr : pub [Field; 2]) {
+    let t = arr.len();
+    constrain t == 2;
+}
+```
+
+## sort
+
+Sorts the array with a built-in function
+
+```rust
+fn sort(_array: Self) -> Self
+```
+
+example
+
+```rust
+// arr = [42, 32]
+fn main(arr : pub [u32; 2]) {
+    let t = arr.sort();
+    constrain t == [32, 42];
+}
+```
+
+## sort_via
+
+Sorts the array with a custom sorting function
+
+```rust
+fn sort_via(mut a: Self, ordering: fn(T, T) -> bool) -> Self
+```
+
+example
+
+```rust
+// arr = [42, 32]
+fn main(arr : pub [u32; 2]) {
+    let t = arr.sort_via(|a, b| a < b);
+    constrain t == [32, 42]; // verifies
+
+    let t = arr.sort_via(|a, b| a > b);
+    constrain t == [32, 42]; // does not verify
```

Comment:
Likewise with variable naming here. Perhaps `sorted_ascending` and `sorted_descending`.